stan/math/rev/arr/functor/coupled_ode_system.hpp

```diff
@@ -45,7 +45,7 @@ struct coupled_ode_system<F, double, var> {
   const F& f_;
   const std::vector<double>& y0_dbl_;
   const std::vector<var>& theta_;
-  const std::vector<double> theta_dbl_;
+  std::vector<var> theta_copy_;
```

Comment:
(For the top-level comment... I can't comment through the PR interface on lines of code that aren't really affected.)

Can you document what this implementation does? I think it'd help anyone else trying to maintain the code.
- that the `theta` is copied and put on the stack, but are not meant to be autodiffed
- that there's use of nested autodiff to compute the sensitivities